.github/workflows/nestbuildmatrix.yml

```diff
@@ -652,12 +651,6 @@ jobs:
         run: |
           mkdir -p "$NEST_VPATH/reports" "$NEST_RESULT"
 
-          if [ "$xNEST_BUILD_COMPILER" = "CLANG" ]; then
-              export CC=clang-11
-              export CXX=clang++-11
-          fi
-          export CXX_FLAGS="-pedantic -Wextra -Wno-unknown-pragmas -D_GLIBCXX_ASSERTIONS"
```

Comment:
Removing this line could make the build a lot less strict than the Linux build. Was this intended?
```bash
export CXX_FLAGS="-pedantic -Wextra -Wno-unknown-pragmas -D_GLIBCXX_ASSERTIONS"
```

Reply:
Here we need more cleanup, I think, since the linux setup also seems inconsistent. For Linux, we first set

https://github.com/nest/nest-simulator/blob/7d9150c0bffd71e3fdce0318d94406303554045e/.github/workflows/nestbuildmatrix.yml#L401-L404

and then later change this to

https://github.com/nest/nest-simulator/blob/7d9150c0bffd71e3fdce0318d94406303554045e/.github/workflows/nestbuildmatrix.yml#L540

Shouldn't we set it in one place only?

Also, why do we have 

https://github.com/nest/nest-simulator/blob/7d9150c0bffd71e3fdce0318d94406303554045e/.github/workflows/nestbuildmatrix.yml#L536-L538

where is `NEST_BUILD_COMPILER` set and why are we binding ourselves to `clang-11`, while the most recent version is 16.0?

Once we have a definite version for Linux, we can use the same for macOS, although we may need to remove `-pedantic` in light of this comment from the old setup:

https://github.com/nest/nest-simulator/blob/03469faebf14f79fb091e170fcc0531ea680acce/build_support/ci_build.sh#L185-L188

Finally and unrelated, what is the purpose of this commented-out section (and the corresponding one for Linux):

https://github.com/nest/nest-simulator/blob/7d9150c0bffd71e3fdce0318d94406303554045e/.github/workflows/nestbuildmatrix.yml#L250

Reply:
You are right, the `CXX_FLAGS` should be set only in one place, preferably the `env:` but probably with the value of the `export`.  In a separate issue we can take a deeper look into `installcheck`, in how far this still is a problem.

The explicit setting of `CC` and `CXX` was necessary at some point because for the clang builds on the runners a wrong compiler was picked up somewhere. This might also be completely obsolete, you are right.

Given that this PR handles the MacOS build, please leave the `build_linux` as is and open a separate issue.

Reply:
OK, tidied up and will create new issue for the Linux part.

Reply:
#2771 is the follow-up for Linux.

Reply:
> You are right, the `CXX_FLAGS` should be set only in one place, preferably the `env:` but probably with the value of the `export`. In a separate issue we can take a deeper look into `installcheck` to see in how far the `-pedantic` still is a problem.
> 
> The explicit setting of `CC` and `CXX` was necessary at some point because for the clang builds on the runners a wrong compiler was picked up somewhere. This might also be completely obsolete, you are right.
> 
> Given that this PR handles the MacOS build, please leave the `build_linux` as is and open a separate issue.

yes, you are right. Explicit setting of CC and CXX was kind of necessary due to the small technical issue like it was picking up a wrong compiler..